pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/ControllerInstanceToggleTest.java

```diff
@@ -16,164 +16,123 @@
 import com.linkedin.pinot.common.utils.helix.HelixHelper;
 import com.linkedin.pinot.controller.helix.core.PinotHelixResourceManager;
 import com.linkedin.pinot.core.query.utils.SimpleSegmentMetadata;
 import java.util.Set;
-import org.apache.helix.manager.zk.ZkClient;
 import org.apache.helix.model.ExternalView;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.testng.Assert;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 
 public class ControllerInstanceToggleTest extends ControllerTest {
-
-  private static final Logger LOGGER = LoggerFactory.getLogger(ControllerInstanceToggleTest.class);
   private static final String HELIX_CLUSTER_NAME = "ControllerInstanceToggleTest";
-  static ZkClient _zkClient = null;
+  private static final String RAW_TABLE_NAME = "testTable";
+  private static final String OFFLINE_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType(RAW_TABLE_NAME);
+  private static final String SERVER_TENANT_NAME = ControllerTenantNameBuilder.getOfflineTenantNameForTenant(null);
+  private static final String BROKER_TENANT_NAME = ControllerTenantNameBuilder.getBrokerTenantNameForTenant(null);
+
+  private final ControllerRequestURLBuilder _requestBuilder =
+      ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL);
 
-  private PinotHelixResourceManager _pinotResourceManager;
+  @Override
+  protected String getHelixClusterName() {
+    return HELIX_CLUSTER_NAME;
+  }
 
   @BeforeClass
   public void setup() throws Exception {
     startZk();
-    _zkClient = new ZkClient(ZkStarter.DEFAULT_ZK_STR);
     startController();
-    _pinotResourceManager = _controllerStarter.getHelixResourceManager();
     ControllerRequestBuilderUtil.addFakeBrokerInstancesToAutoJoinHelixCluster(HELIX_CLUSTER_NAME,
-        ZkStarter.DEFAULT_ZK_STR, 20, true);
+        ZkStarter.DEFAULT_ZK_STR, 5, true);
     ControllerRequestBuilderUtil.addFakeDataInstancesToAutoJoinHelixCluster(HELIX_CLUSTER_NAME,
-        ZkStarter.DEFAULT_ZK_STR, 20, true);
-  }
-
-  @AfterClass
-  public void tearDown() {
-    stopController();
-    try {
-      if (_zkClient.exists("/" + HELIX_CLUSTER_NAME)) {
-        _zkClient.deleteRecursive("/" + HELIX_CLUSTER_NAME);
-      }
-    } catch (Exception e) {
-    }
-    _zkClient.close();
-    stopZk();
+        ZkStarter.DEFAULT_ZK_STR, 5, true);
   }
 
   @Test
-  public void testInstanceToggle()
-      throws Exception {
-    // Create offline table creation request
-    String tableName = "testTable";
+  public void testInstanceToggle() throws Exception {
+    // Create an offline table
     String tableJSONConfigString =
-        new TableConfig.Builder(CommonConstants.Helix.TableType.OFFLINE).setTableName(tableName)
-            .setNumReplicas(20)
+        new TableConfig.Builder(CommonConstants.Helix.TableType.OFFLINE).setTableName(RAW_TABLE_NAME)
+            .setNumReplicas(5)
```

Comment:
Make this a constant, so if you change the number of instances, it gets reflected here as well.